src/codegen/debug.rs

```diff
@@ -351,9 +416,208 @@ impl<'ink> DebugBuilder<'ink> {
 
         Ok(())
     }
+
+    fn create_debug_location(
+        &self,
+        scope: DIScope<'ink>,
+        line: usize,
+        column: usize,
+    ) -> Result<DILocation, Diagnostic> {
+        Ok(self.debug_info.create_debug_location(
+            self.context,
+            line as u32,   // try_into() error msg on fail
+            column as u32, // not implemented yet
+            scope,
+            None, // ?
+        ))
+    }
+
+    fn create_subroutine_type(
+        &self,
+        return_type: Option<&DataType>,
+        parameter_types: &[&DataType],
+    ) -> Result<DISubroutineType, Diagnostic> {
```

Comment:
result not required, except if you return an Err instead of panicing at line 451